src/parameters.rs

```diff
@@ -0,0 +1,334 @@
+    ///
+    /// EXAMPLE:
+    /// SystemIdentity {
+    /// id: "2e483fe9",
+    /// name: Some("test1"),
+    /// client_id: "8dbf3d32",
+    /// organization_id: "b961cf81",
+    /// identity_type: L1 {
+    /// client_secret: "AfYFAUjf9",
+    /// credential_expiration: "2025-06-04T19:25:00Z"
+    /// }
+    /// }
+    Secret(SecretArgs),
+    /// Creates an identity whose type is 'private key' or known as a parent identity.
+    /// This type of identity does not expire.
+    ///
+    /// EXAMPLE:
+    /// SystemIdentity(
+    /// id: e5af42f2,
+    /// name: test,
+    /// client_id: 8150a0ee,
+    /// organization_id: b961cf81,
+    /// identity_type: L2(pub_key: LS0tLS1)
+    /// )
+    Key(KeyArgs),
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct KeyArgs {
+    /// Basic information need for auth name, client_id, etc.
+    #[command(flatten)]
+    basic_auth_args: BasicAuthArgs,
+
+    /// Add the access token for identity creation, only bearer token type is accepted
+    #[arg(long)]
+    bearer_access_token: String,
+
+    /// Options for configuring the output destination (required for Key identity)
+    #[command(flatten)]
+    output_options: OutputDestinationArgs,
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct SecretArgs {
+    /// Basic information need for auth name, client_id, etc.
+    #[command(flatten)]
+    basic_auth_args: BasicAuthArgs,
+
+    /// Add the access token for identity creation, only bearer token type is accepted
+    #[arg(long)]
+    bearer_access_token: String,
+}
+
+#[derive(ValueEnum, Clone, Debug, PartialEq)]
+pub enum OutputPlatformChoice {
+    #[value(name = "local-file")]
+    LocalFile,
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct OutputDestinationArgs {
+    /// Platform for the output of the generated key or resource.
+    #[arg(long, short, value_enum)]
+    output_platform: OutputPlatformChoice,
+
+    /// File path where the private key output will be saved (required if --output-platform=local-file).
+    #[arg(long, short)]
+    output_local_path: Option<PathBuf>,
+}
+
+pub fn create_metadata_for_token_retrieve(
+    client_id: String,
+    environment: Environments,
+    auth_args: &AuthInputArgs,
+    endpoints: ExternalEndpoints,
+) -> Result<SystemTokenCreationMetadata, Box<dyn std::error::Error>> {
+    let auth_method = select_auth_method(
+        auth_args.client_secret.clone(),
+        auth_args.private_key_path.clone(),
+    )?;
+    let environment = select_environment(
+        environment,
+        endpoints.token_renewal_endpoint,
+        endpoints.system_identity_creation_endpoint,
+    )?;
+
+    Ok(SystemTokenCreationMetadata {
+        client_id,
+        environment,
+        auth_method,
+    })
+}
+pub fn create_metadata_for_identity_creation(
+    identity_type: IdentityType,
+) -> Result<SystemIdentityCreationMetadata, Box<dyn std::error::Error>> {
+    let (basic_auth_args, _json_token, output_platform, _output_local_path) = match identity_type {
```

Comment:
Why do we need to return `_json_token` and `_output_local_path` if they are not used? Could we simplify and return two items only?